src/uct/base/uct_md.c

```diff
@@ -418,20 +423,18 @@ ucs_status_t uct_mem_alloc_check_params(const uct_mem_alloc_params_t *params)
          method < (params->methods.methods + params->methods.count);
          ++method) {
 
-        if (*method == UCT_ALLOC_METHOD_MD) {
-
-            if (params->mds.count == 0) {
-                ucs_debug("count of mds provided is set to zero: %s",
-                          ucs_status_string(UCS_ERR_INVALID_PARAM));
-                return UCS_ERR_INVALID_PARAM;
-            }
+        if ((params->methods.count == 1) && (method[0] == UCT_ALLOC_METHOD_MD) &&
+            (params->mds.count == 0)) {
```

Comment:
maybe drop this, the main function would fail anyway if no method is set

Reply:
But this check catches the case when method is set to METHOD_MD but user forgot to populate params->mds.mds and params->mds.count. Should we not continue to keep this?